src/btree/bt_cursor.c

```diff
@@ -300,21 +300,18 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
      * update that's been deleted is not a valid key/value pair).
      */
     if (cbt->ins != NULL) {
-        if (WT_CURSOR_BOUNDS_SET(&cbt->iface)) {
+        if (btree->type == BTREE_ROW && WT_CURSOR_BOUNDS_SET(&cbt->iface) && key == NULL) {
```

Comment:
Why do we no longer check `WT_CURSOR_BOUNDS_SET(&cbt->iface)` before executing any of lines 303-314? Is this to clean up the if statement here?